packages/flutter/lib/src/painting/text_span.dart

```diff
@@ -356,25 +356,24 @@ class TextSpan extends InlineSpan implements HitTestTarget, MouseTrackerAnnotati
   }
 
   /// Returns the text span that contains the given position in the text.
-  @override
-  InlineSpan? getSpanForPositionVisitor(TextPosition position, Accumulator offset) {
-    final String? text = this.text;
-    if (text == null || text.isEmpty) {
-      return null;
-    }
-    final TextAffinity affinity = position.affinity;
-    final int targetOffset = position.offset;
-    final int endOffset = offset.value + text.length;
-
-    if (offset.value == targetOffset && affinity == TextAffinity.downstream ||
-        offset.value < targetOffset && targetOffset < endOffset ||
-        endOffset == targetOffset && affinity == TextAffinity.upstream) {
-      return this;
-    }
-    offset.increment(text.length);
-    return null;
+@override
+InlineSpan? getSpanForPositionVisitor(TextPosition position, Accumulator offset) {
+  final int currentOffset = offset.value;
```

Comment:
I'm not sure the changes in this method are necessary (the issue seems to be fixed with only the `WidgetSpan` changes). Can you explain what cases the changes in this code are trying to solve?